ompi/mca/osc/monitoring/configure.m4

```diff
@@ -0,0 +1,89 @@
+# -*- shell-script -*-
+#
+# Copyright (c) 2016-2017 Inria.  All rights reserved.
+# $COPYRIGHT$
+#
+# Additional copyrights may follow
+#
+# $HEADER$
+#
+
+# mca_ompi_osc_monitoring_generate_templates
+#
+# Append to file $1 an array of components by listing component names in $2.
+#
+# $1 = filename
+# $2 = osc component names
+#
+AC_DEFUN([mca_ompi_osc_monitoring_generate_templates],[
+    m4_ifval([$1], [
+        cat <<EOF >[$1]
+/*
+ * DO NOT EDIT - This file generated from
+ * ompi/mca/osc/monitoring/configure.m4
+ */
+/*
+ * Copyright (c) 2017      Inria.  All rights reserved.
+ * \$COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * \$HEADER$
+ */
+
+#ifndef MCA_OSC_MONITORING_GEN_TEMPLATE_H
+#define MCA_OSC_MONITORING_GEN_TEMPLATE_H
+
+#include <string.h>
+#include <ompi_config.h>
+#include <ompi/mca/osc/osc.h>
+#include "osc_monitoring_template.h"
+
+/***************************************/
+/* Include template generating macros  */
+
+EOF
+        # Generate each case in order to register the proper template functions
+        for comp in `echo $2 | sed -e 's/,/ /g' -e 's/monitoring//'`
+        do
+            echo "OSC_MONITORING_MODULE_TEMPLATE_GENERATE(${comp})" >>[$1]
+        done
+        cat <<EOF >>[$1]
+
+/***************************************/
+
+static inline int
+ompi_mca_osc_monitoring_set_template(ompi_osc_base_component_t *best_component,
+                                     ompi_osc_base_module_t *module)
+{
+EOF
+        echo -n '    ' >>[$1]
+        for comp in `echo $2 | sed -e 's/,/ /g' -e 's/monitoring//'`
+        do
+            echo "if ( 0 == strcmp(\"${comp}\", best_component->osc_version.mca_component_name) ) {" >>[$1]
+            echo "        OSC_MONITORING_SET_TEMPLATE(${comp}, module);" >>[$1]
+            echo -n '    } else ' >>[$1]
+        done
+        echo '' >>[$1]
+        cat <<EOF >>[$1]
+        return OMPI_ERR_NOT_SUPPORTED;
```

Comment:
I have to say that this shell script is really ugly to read.  It would be a lot clearer (in my opinion) if you didn't try to get the else if on the same line and just had some slightly uglier code generated.

Reply:
I have to say I totally agree with you :-D And it actually led to issue with OS-X, and replacing "echo -n" with "printf" did not worked either. Commit 2011660bea186e303a9f46f5591d19838343abc2 fixes the issue. And at this point I start feeling like I should not have kept the wrong commits, and only keep once everything was fixed. I'll do that tomorrow.